### ISSUE TITLE
Plugin need: add hook: template:task:* and controller:task:*

### ISSUE THREAD

**Issue by oliviermaridat** [fguillot/kanboard#1742]
Even if I would like to find a proper solution to add #1009 (Milestone management) into Kanboard, I intend to provide it as a plugin to have time to make it work properly.

For this I will need to update the task view. That is why I will need a template hook like: template:task:before-tasklink. Is it possible to add it please?

Actually, if there is no performance cost, I propose to add the following template hooks to be able to put something everywhere on this view:

* template:task:top
* template:task:before-timeedit
* template:task:before-description
* template:task:before-tasklinks
* template:task:before-subtasks
* template:task:before-timetracking
* template:task:before-attachements
* template:task:before-comments
* template:task:bottom

I will also need to remove a type of link in the task link view. This may require a controller:task:links hook, but I am not very confident about how things work there...

Thank you!

**Comment by BlueTeck** [comment]
:+1: I also thought about these hooks.

**Comment by oliviermaridat** [comment]
Ok, I have made some tests and this is now clearer for me.

Here the plugin (work in progress): https://github.com/oliviermaridat/kanboard-milestone-plugin
Here is a commit for Kanboard to add one controller and one template hooks: https://github.com/oliviermaridat/kanboard/commit/07d870e1cb31fdc9733a46c29336df82eca1de74

I have added the template hook "template:task:before-tasklink" in the task view, and now I can add code there. Good.

I have also added a merged controller hook "controller:task:show:params" in Controller\Task::show that allows me to modify the parameters sent to the task view. With that I managed to update the "links" parameter as I want. I also add a "milestone_links" parameter, but I can't use it in my custom template (hooking the new "template:task:before-tasklink") because I don't have access to anything there...

So the question is: how can I add new parameter in an existing controller ouput, and how can I use it in a new custom template?
